x/bank/types/genesis.go

```diff
@@ -58,22 +58,28 @@ func (gs GenesisState) Validate() error {
 		}
 	}
 
+	for _, addr := range gs.BlockedAddrs {
+		if _, err := sdk.AccAddressFromBech32(addr); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
 // NewGenesisState creates a new genesis state.
-func NewGenesisState(params Params, balances []Balance, supply sdk.Coins, denomMetaData []Metadata) *GenesisState {
+func NewGenesisState(params Params, balances []Balance, supply sdk.Coins, denomMetaData []Metadata, blockedAddrs []string) *GenesisState {
```

Comment:
How about accepting `blockedAddrs []sdk.AccAddress`?

Reply:
Ref: https://github.com/cosmos/cosmos-sdk/blob/4081bc3d0080644541ba3de34fbe1bde27779565/x/distribution/types/genesis.go#L9

Reply:
BankKeeper and others are receiving BlockedAddr as string, so string was used to follow this format as much as possible.

Ref. https://github.com/cosmos/cosmos-sdk/blob/4081bc3d0080644541ba3de34fbe1bde27779565/x/bank/keeper/keeper.go%23L96-L110

Reply:
I think the original version used `string`, not `sdk.AccAddress` because it is used as key of a map.
Since `sdk.AccAddress` is `[]byte`, the original author might thought that it's not suitable for map keys.
After changing the parameter type from map to slice, we can safely use `[]sdk.AccAddress` again.
But as you said, to keep consistency between the original one, it's still possible to use `[]string` as well.